token/js/src/extensions/extensionType.ts

```diff
@@ -28,6 +29,7 @@ export enum ExtensionType {
     CpiGuard,
     PermanentDelegate,
     NonTransferableAccount,
+    TransferHook,
```

Comment:
Can you also add `TransferHookAccount` at https://github.com/solana-labs/solana-program-library/blob/47f09c903f567bfee1e9d5a1e9ab83770208449e/token/program-2022/src/extension/mod.rs#L894-L895 and wire it up properly as an account requirement for `TransferHook`? Otherwise we won't be able to create token accounts for mints that have transfer hooks.